kaggle-classification/keras_trainer/rnn.py

```diff
@@ -35,10 +35,11 @@ def get_model(self):
         trainable=self.hparams.train_embedding)(
             I)
     H = Bidirectional(GRU(128, return_sequences=True))(E)
-    A = TimeDistributed(Dense(3), input_shape=(sequence_length, 256))(H)
-    A = Flatten()(A)
-    A = Dense(sequence_length, activation='softmax')(A)
+    A = TimeDistributed(
+        Dense(1, activation='softmax'), input_shape=(sequence_length, 256))(
```

Comment:
This looks good. I'd actually add two dense layers (say one of size 128 and then one of size 1). That way the neurons in the output of a single lstm timestep get to interact before deciding how much attention to pay. Does not have to be an immediate fix.